homeassistant/auth/providers/__init__.py

```diff
@@ -149,3 +159,86 @@ def async_create_credentials(self, data: Dict[str, str]) -> Credentials:
 
     processed.add(provider)
     return module
+
+
+class LoginFlow(data_entry_flow.FlowHandler):
+    """Handler for the login flow."""
+
+    def __init__(self, auth_provider: AuthProvider):
+        """Initialize the login flow."""
+        self._auth_provider = auth_provider
+        self._auth_module_id = None
+        self._auth_manager = auth_provider.hass.auth
+        self.available_mfa_modules = []
+        self.created_at = dt_util.utcnow()
+        self.user = None
+
+    async def async_step_init(self, user_input=None):
+        """Handle the first step of login flow.
+
+        Return self.async_show_form(step_id='init') if user_input == None.
+        Return await self.async_start_mfa(flow_result) if login init step pass.
```

Comment:
Where is `async_start_mfa` defined?

Reply:
I would expect `async_start_mfa` to not do anything if the context is `type=link_user`.

Reply:
Aha, that is from old design, missed to update document. It should change to 
Return await self.async_finish(flow_result) if login init step pass